server/tests/api/roles.create.test.js

```diff
@@ -0,0 +1,140 @@
+import 'dotenv/config';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app';
+
+import { testTokens } from '../data/api.roles.sample';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('*** Testing Roles API: POST new role ***', () => {
+  // -------------------------------------
+  // Hooks: https://mochajs.org/#hooks
+  // -------------------------------------
+
+  before(async () => {});
+  after(async () => {});
+
+  // -------------------------------------
+  // Test Cases
+  // -------------------------------------
+
+  // Role exists, and User has correct permissions
+  it('should create a new Role', (done) => {
+    chai
+      .request(app)
+      .post('/api/v1/roles')
+      .set('Authorization', testTokens.valid)
+      .send({
+        name: 'Test Role',
+        permissions: 'test-permission',
+      })
+      .end((err, res) => {
+        expect(res.statusCode).to.equal(201);
+        expect(res.type).to.equal('application/json');
+        expect(res.body).to.have.property('status');
+        expect(res.body).to.have.property('message');
+        expect(res.body).to.have.property('data');
+        expect(res.body.data).to.have.property('name');
+        expect(res.body.data).to.have.property('permissions');
+
+        expect(res.body.status).to.equal(201);
```

Comment:
@fordarnold when testing APIs with 200 or 201, you can go one step further and make sure that the data (values of the response properties) are actually what have been sent or expected to be returned. In that case, the tests are more robust as they test contents both key & values of the response object.

Reply:
Yes, I believe I did so... see lines 44-47

```
expect(res.body.data).to.eql({
  name: 'Test Role',
  permissions: 'test-permission',
});
```